src/test/java/emissary/core/IBaseDataObjectHelperTest.java

```diff
@@ -0,0 +1,823 @@
+
+        IBaseDataObjectHelper.diff(ibdo1, ibdo1, differences, false, false, false, true);
+
+        Assertions.assertEquals(0, differences.size());
+
+        IBaseDataObjectHelper.diff(ibdo1, ibdo2, differences, false, false, false, true);
+
+        Assertions.assertEquals(1, differences.size());
+    }
+
+    @Test
+    void testDiffParameters() {
+        ibdo1.putParameter("STRING", "string");
+        ibdo1.putParameter("LIST", Arrays.asList("first", "second", "third"));
+
+        IBaseDataObjectHelper.diff(ibdo1, ibdo1, differences, false, false, false, false);
+
+        Assertions.assertEquals(0, differences.size());
+
+        IBaseDataObjectHelper.diff(ibdo1, ibdo2, differences, false, false, false, false);
+
+        Assertions.assertEquals(1, differences.size());
+    }
+
+    @Test
+    void testDiffAlternateViews() {
+        ibdo1.addAlternateView("AAA", "AAA".getBytes(StandardCharsets.US_ASCII));
+        ibdo1.addAlternateView("BBB", "BBB".getBytes(StandardCharsets.US_ASCII));
+        ibdo1.addAlternateView("CCC", "CCC".getBytes(StandardCharsets.US_ASCII));
+
+        IBaseDataObjectHelper.diff(ibdo1, ibdo1, differences, false, false, false, false);
+
+        Assertions.assertEquals(0, differences.size());
+
+        IBaseDataObjectHelper.diff(ibdo1, ibdo2, differences, false, false, false, false);
+
+        Assertions.assertEquals(1, differences.size());
+
+        ibdo2.addAlternateView("DDD", "DDD".getBytes(StandardCharsets.US_ASCII));
+        ibdo2.addAlternateView("EEE", "EEE".getBytes(StandardCharsets.US_ASCII));
+        ibdo2.addAlternateView("FFF", "FFF".getBytes(StandardCharsets.US_ASCII));
+
+        differences.clear();
+
+        IBaseDataObjectHelper.diff(ibdo1, ibdo2, differences, false, false, false, false);
+
+        Assertions.assertEquals(1, differences.size());
+    }
+
+    @Test
+    void testDiffPriority() {
+        ibdo1.setPriority(13);
+
+        IBaseDataObjectHelper.diff(ibdo1, ibdo1, differences, false, false, false, false);
+
+        Assertions.assertEquals(0, differences.size());
+
+        IBaseDataObjectHelper.diff(ibdo1, ibdo2, differences, false, false, false, false);
+
+        Assertions.assertEquals(1, differences.size());
+    }
+
+    @Test
+    void testDiffCreationTimestamp() {
+        ibdo1.setCreationTimestamp(new Date(1234567890));
+
+        IBaseDataObjectHelper.diff(ibdo1, ibdo1, differences, false, true, false, false);
+
+        Assertions.assertEquals(0, differences.size());
+
+        IBaseDataObjectHelper.diff(ibdo1, ibdo2, differences, false, true, false, false);
+
+        Assertions.assertEquals(1, differences.size());
+    }
+
+    @Test
+    void testDiffExtractedRecords() {
+        ibdo1.addExtractedRecord(new BaseDataObject());
+        ibdo1.addExtractedRecord(new BaseDataObject());
+        ibdo1.addExtractedRecord(new BaseDataObject());
+
+        IBaseDataObjectHelper.diff(ibdo1, ibdo1, differences, false, false, false, false);
```

Comment:
Not a blocker, but these next 7 lines should be refactored into a simple method since they're repeated MANY times.
```
        IBaseDataObjectHelper.diff(ibdo1, ibdo1, differences, false, false, false, false);
        
        Assertions.assertEquals(0, differences.size());

        IBaseDataObjectHelper.diff(ibdo1, ibdo2, differences, false, false, false, false);

        Assertions.assertEquals(1, differences.size());
```

Reply:
:+1: Fixed in next revision